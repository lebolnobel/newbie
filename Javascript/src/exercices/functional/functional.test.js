import {
  numbers,
  sum,
  notHigherThan50,
  checkIfAtLeastOneElementIsHigherThan10000,
  divideByTwoThenCheckIfNumbersAreLowerThan500,
  getFirstElementHigherThan50
} from "./functional";

describe("Functional programming", () => {
  it("should return sum elements", () => {
    expect(sum(numbers)).toMatchSnapshot();
  });

  it("should return elements not higher than 50", () => {
    expect(notHigherThan50(numbers)).toMatchSnapshot();
  });

  it("should return true if at least one element is higher than 10 000", () => {
    expect(
      checkIfAtLeastOneElementIsHigherThan10000(numbers)
    ).toMatchSnapshot();
  });

  it("should return if all numbers divided by two are lower than 500", () => {
    expect(
      divideByTwoThenCheckIfNumbersAreLowerThan500(numbers)
    ).toMatchSnapshot();
  });

  it("should return the first element higher than 500", () => {
    expect(getFirstElementHigherThan50(numbers)).toMatchSnapshot();
  });
});
