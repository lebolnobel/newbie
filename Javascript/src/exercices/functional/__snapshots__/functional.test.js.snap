// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Functional programming should return elements not higher than 50 1`] = `
Array [
  1,
  32,
  45,
  45,
  -23,
  45,
  0,
]
`;

exports[`Functional programming should return if all numbers divided by two are lower than 500 1`] = `true`;

exports[`Functional programming should return sum elements 1`] = `1909`;

exports[`Functional programming should return the first element higher than 500 1`] = `783`;

exports[`Functional programming should return true if at least one element is higher than 10 000 1`] = `false`;
